res/controllers/Behringer-CMDStudio4a-scripts.js

```diff
@@ -0,0 +1,257 @@
+// ****************************************************************************
+// * Mixxx mapping script file for the Behringer CMD Studio 4a.
+// * Author: Craig Easton
+// * Version 1.2 (Jan 2016)
+// * Forum: http://www.mixxx.org/forums/viewtopic.php?f=7&amp;t=7868
+// * Wiki: http://www.mixxx.org/wiki/doku.php/behringer_cmd_studio_4a
+// ****************************************************************************
+
+////////////////////////////////////////////////////////////////////////
+// JSHint configuration                                               //
+////////////////////////////////////////////////////////////////////////
+/* global engine                                                      */
+/* global script                                                      */
+/* global print                                                       */
+/* global midi                                                        */
+//////////////////////////////////////////////////////////////////////// 
+
+// Master function definition.
+function BehringerCMDStudio4a() {}
+
+
+// ***************************** Global Vars **********************************
+
+// Shift/mode state variables.
+BehringerCMDStudio4a.delState = {0:false,1:false,2:false,3:false};
+BehringerCMDStudio4a.scratchState = {0:false,1:false,2:false,3:false};
+
+// Button push/release state variables.
+BehringerCMDStudio4a.pitchDec = {0:false,1:false,2:false,3:false};
+BehringerCMDStudio4a.pitchInc = {0:false,1:false,2:false,3:false};

```

Comment:
It would make more sense to declare these variables as arrays. For example:
``BehringerCMDStudio4a.pitchInc = [false,false,false,false];